test/scripts/products/h22-validate-that-rate-limit-service-is-working-as-expected/h22-validate-that-rate-limit-service-is-working-as-expected.go

```diff
@@ -82,13 +93,14 @@ func main() {
 	fmt.Printf("ℹ️  Redis host: %s\n", redisHost)
 
 	// Create the threescale API
-	api, err := createAPI(ctx, client, namespacePrefix, "h22-test")
+	api, err := createAPI(ctx, client, namespacePrefix, baseName)
 	if err != nil {
 		fmt.Println(err)
+		deleteAPI(cleanupAPI, api, ctx, client, namespacePrefix)
 		os.Exit(1)
 	}
 	fmt.Printf("️🔌  Created API. Endpoint %s\n", api.Endpoint)
-	defer deleteAPI(cleanupAPI, api)
+	defer deleteAPI(cleanupAPI, api, ctx, client, namespacePrefix)
 
 	// Wait for the Redis Pod to finish creating
 	redisPod := <-redisCreated
```

Comment:
Should `cleanupRedis(ctx, client, namespace)` be called in the `if` block if `err != nil`?

Reply:
Just to clarify, this comment applies to lines [107-110](https://github.com/integr8ly/integreatly-operator/pull/2372/files#diff-ff13b5fdbdad9451b9415e9bfbceac31e8b17a763eedb2ad3aa9a5a53c2aa68cR107-R110) but GitHub doesn't allow you to add comments to collapsed sections.

Reply:
you have a point. I've created an `osExit1()` func to be called in such places - it will clean up all the resources